### PR TITLE
feat: back-end of glassPart

### PR DESCRIPTION
**Pull request by No3ml3**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [ ] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [x] I tested admin by manually adding content from zero
- [x] I created easy to use admin experience which is self-explanatory
- [x] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by jaalazard — DISMISSED** [review]

File: packages/backend/api/src/glass.ts
Comment (Justicesvr) [inline packages/backend/api/src/glass.ts:1]: random pour les verres ok

**Review by Justicesvr — DISMISSED** [review]

File: packages/backend/api/src/router.ts [hunk 1]
```diff
@@ -43,5 +44,6 @@ router.get('/alcohols/:level', async (req, res) => {
 
 router.use('/auth', authRouter);
 router.use('/ingredient', ingredient);
+router.use('/', glass);
```
Comment (YDrogen) [inline packages/backend/api/src/router.ts]: Il vaut mieux mettre l'URL ici

File: packages/backend/api/src/glass.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import express from 'express';
+import { sql } from 'kysely';
+
+import { db } from '@app/backend-shared';
+
+const glass = express.Router();
+
+glass.get('/glass', async (req, res) => {
```
Comment (YDrogen) [inline packages/backend/api/src/glass.ts]: plutôt qu'ici

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+import { useEffect, useState } from 'react';
 
-import type { GlassPartProps } from '@app/types';
+import type { Glass, GlassPartProps } from '@app/types';
+
+import useFetch from '@/hooks/use-fetch';
+
+export default function GlassPart({ errors, setValue, watch }: GlassPartProps) {
+  const [isReload, setIsReload] = useState(false);
+  const url = `${import.meta.env.VITE_API_URL}/glass`;
```
Comment (YDrogen) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx]: À mettre hors du component

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx [hunk 4]
```diff
@@ -1,8 +1,24 @@
...
+  const { data } = useFetch<Pick<Glass, 'name' | 'id'>[]>(url, [isReload]);
+
+  const shuffleClick = () => {
+    setIsReload(!isReload);
+  };
```
Comment (YDrogen) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx]: Pourquoi ne pas faire un fetch ici ? Plutôt que faire un state qui sert à faire croire au useEffect du useFetch que quelque chose à changé alors que non ?

File: packages/frontend/web/src/hooks/use-fetch.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ export default function useFetch<T>(url: string) {
     return () => {
       controller.abort();
     };
-  }, [url]);
+  }, [url, ...deps]);
```
Comment (YDrogen) [inline packages/frontend/web/src/hooks/use-fetch.ts]: Très mauvaise idée, on ne contrôle pas ce qui va déclencher le useEffect

File: packages/types/src/cocktail-form.ts [hunk 6]
```diff
@@ -42,8 +42,12 @@ export interface IngredientsPartProps {
 }
 
 export interface GlassPartProps {
-  register: UseFormRegister<CocktailForm>;
   errors: FieldErrors<CocktailForm>;
+  setValue: (
+    name: keyof CocktailForm,
+    value: string | { id: number; name: string },
+  ) => void;
```
Comment (YDrogen) [inline packages/types/src/cocktail-form.ts]: Comme pour le typage du `watch` "`UseFormWatch`", tu as aussi un type pour le `setValue`

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by jaalazard — DISMISSED** [review]

**Review by Justicesvr — APPROVED** [review]

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx:1]: plus de usefetch :(

**Review by Justicesvr — DISMISSED** [review]

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx (hunk 4 above)
Comment (YDrogen) [reply packages/frontend/web/src/components/form-cocktail/GlassPart.tsx]: Je repose la question, pourquoi ne pas faire un simple fetch ici quand tu cliques sur le bouton ?
Pourquoi absolument vouloir passer par un `useEffect` ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx (hunk 4 above)
Comment (No3ml3) [reply packages/frontend/web/src/components/form-cocktail/GlassPart.tsx]: les consignes n'étaient pas claire

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx:1]: ok plus de useEffect

**Review by Justicesvr — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx [hunk 7]
```diff
@@ -1,8 +1,33 @@
...
+    controller.abort();
+    const newController = new AbortController();
+    setController(newController);
+
+    try {
+      await fetchData(url, newController);
+    } catch (error) {
+      console.error(error);
+    }
+  };
```
Comment (YDrogen) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx]: Est-ce que tu comprends ce que tu fais ?
- Pourquoi il n'y a plus de `useEffect` ni de `useFetch` ? Comment est chargé le premier verre ?
- Pourquoi il y a un `AbortController` ? Il sert à quoi ici ?
- Pourquoi ça ne dérange que moi ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by jaalazard — DISMISSED** [review]

**Review by Cardamced — DISMISSED** [review]

File: packages/frontend/web/src/components/form-cocktail/GlassPart.tsx
Comment (YDrogen) [inline packages/frontend/web/src/components/form-cocktail/GlassPart.tsx:1]: Impec 👍 
Dernière question cependant : Pourquoi le backend envoi une array si tu n'utilise que le premier élément ? `data[0]` Pourquoi ne pas renvoyer qu'un seul élément ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by jaalazard — APPROVED** [review]

File: packages/backend/api/src/glass.ts
Comment (Justicesvr) [inline packages/backend/api/src/glass.ts:1]: le fameux select dans un select

**Review by Justicesvr — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**